src/index.js

```diff
@@ -32,6 +35,16 @@ function loader (css, map, meta) {
 
   const sourceMap = options.sourceMap
 
+  let prevAst = null
+  if (
+    meta &&
+    meta.ast &&
+    meta.ast.type === 'postcss' &&
+    satisfies(meta.ast.version, `^${postcssPkg.version}`)
```

Comment:
What/Which format does `meta.ast.version` currently return? I think the `major` should be enough as it is unlinkely to have AST changes without introducing a BREAKING CHANCE. Needs to be documented somehow in any case.

Assumming to return the `major` version ist schould be sufficient to e.g 

```js
if (
  meta &&
  meta.ast &&
  meta.ast.type === 'postcss' &&
  meta.ast.version === version.major
) ...
```

I also think the `meta` property is always defined, but not 100% sure anymore...

Reply:
it's the full version string from the postcss Processor. hopefully just the major version is enough to determine breaking changes, but ppl and computers are messy and breaking changes  in AST's happen whenever in practice. Yes this a "best guess", but i think it's wise to encode more rather than less information here since we know it. It means that down the road if something does break unexpectedly and we (or consumers of this metadata downstream) need to adjust on a minor or patch version it would be possible, and not require a breaking API change here to add

Reply:
I will open an issue in the postcss repo maybe it's possible to add the `version` to e.g the `ast.root` so any third party tool does have to invent it's own way to handle this (not blocking this PR on it) but maybe we are luky...

Reply:
oo great idea